### PR TITLE
[esp/bindings_js] fix multiple bugs in semantic segmentation

### PR DESCRIPTION
**Pull request by msbaines**
* shape of semantic observation was 4x too big due to channels
  property of sensor spec defaulting to 4
* semantic sensor data is flipped-Y from RGB image
  now that we render the RGB directly
* we were only using least significat 8-bits of object_id
  causing wrong category to be displayed for many objects
* add support for pre-loading replica's info_semantic.json
  * needs PR #250 to be usable

Co-Authored-By: Erik Wijmans <etw@gatech.edu>

## Motivation and Context

Fix number of bugs in WebGL semantic information.

## How Has This Been Tested

replica apartment_0 scene in WebGL

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by msbaines** [comment]
Ready for review.

File: src/esp/bindings_js/modules/navigate.js [hunk 1]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
 
     const obs = this.sim.getObservation("semantic", null);
     this.semantic_data = obs.getData();
-    let data = this.semantic_data;
+    const object_ids = new Uint32Array(
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]:
```suggestion
    const objectIds = new Uint32Array(
```

File: src/esp/bindings_js/modules/navigate.js [hunk 2]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
...
+      this.semantic_data.buffer,
+      this.semantic_data.byteOffset,
+      this.semantic_data.length / 4
+    );
 
     // TOOD(msb) implement a better colorization scheme
-    for (let i = 0; i < 640 * 480; i++) {
-      if (data[i * 4] & 1) {
+    for (let i = 0; i < object_ids.length; i++) {
+      const object_id = object_ids[i];
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]:
```suggestion
      const objectId = objectIds[i];
```

File: src/esp/bindings_js/modules/navigate.js [hunk 3]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
...
+    const object_ids = new Uint32Array(
+      this.semantic_data.buffer,
+      this.semantic_data.byteOffset,
+      this.semantic_data.length / 4
+    );
 
     // TOOD(msb) implement a better colorization scheme
-    for (let i = 0; i < 640 * 480; i++) {
-      if (data[i * 4] & 1) {
+    for (let i = 0; i < object_ids.length; i++) {
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]:
```suggestion
    for (let i = 0; i < objectIds.length; i++) {
```

File: src/esp/bindings_js/modules/navigate.js [hunk 4]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
...
+      const object_id = object_ids[i];
+      if (object_id & 1) {
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]:
```suggestion
      if (objectId & 1) {
```

File: src/esp/bindings_js/modules/navigate.js [hunk 5]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
...
         this.semanticImageData.data[i * 4] = 255;
       } else {
         this.semanticImageData.data[i * 4] = 0;
       }
-      if (data[i * 4] & 2) {
+      if (object_id & 2) {
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]:
```suggestion
      if (objectId & 2) {
```

File: src/esp/bindings_js/modules/navigate.js [hunk 6]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
...
         this.semanticImageData.data[i * 4 + 1] = 255;
       } else {
         this.semanticImageData.data[i * 4 + 1] = 0;
       }
-      if (data[i * 4] & 4) {
+      if (object_id & 4) {
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]:
```suggestion
      if (objectId & 4) {
```

File: src/esp/bindings_js/modules/navigate.js [hunk 7]
```diff
@@ -75,21 +80,26 @@ class NavigateTask {
 
     const obs = this.sim.getObservation("semantic", null);
     this.semantic_data = obs.getData();
```
Comment (apsdehal) [inline src/esp/bindings_js/modules/navigate.js]: Also  make `this.semanticData` everywhere.

**Comment by apsdehal** [comment]
Changes look good overall.

**Pull request merged**